plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -322,24 +322,34 @@ public function authorize_or_capture_payment( $order, $action_url, $action = WC_
 	 * @throws Exception If the PayPal payment capture fails.
 	 */
 	public function capture_authorized_payment( $order ) {
-		if ( ! $order || ! $order->get_transaction_id() ) {
-			WC_Gateway_Paypal::log( 'PayPal authorization ID not found. Cannot capture payment.' );
+		$authorization_id = $order->get_meta( '_paypal_authorization_id', true );
+		$paypal_order_id  = $order->get_meta( '_paypal_order_id', true );
+		$paypal_status    = $order->get_meta( '_paypal_status', true );
+
+		if ( ! $order || ! $paypal_order_id ) {
```

Comment:
We're doing a number of operations against `$order` before validating that it exists. 😬 
```suggestion
		if ( ! $order ) {
			return; // And maybe log something?
		}

		$authorization_id = $order->get_meta( '_paypal_authorization_id', true );
		$paypal_order_id  = $order->get_meta( '_paypal_order_id', true );
		$paypal_status    = $order->get_meta( '_paypal_status', true );

		if ( ! $paypal_order_id ) {
```